Update Cody capabilities (#56269)
sourcegraph/sourcegraph: doc/cody/overview/index.md

## What is Cody?

> Cody is a free and open-source AI coding assistant that can write, understand, and fix code. Cody is powered by Sourcegraph's <a href="https://docs.sourcegraph.com/cody/explanations/code_graph_context">code graph</a> and uses knowledge of your entire codebase to give more accurate answers and generate idiomatic code.

Reviewer: ```suggestion
Cody is a free, open-source AI coding assistant who writes, understands, and fixes code. Cody is powered by Sourcegraph's <a href="https://docs.sourcegraph.com/cody/explanations/code_graph_context">code graph</a> and uses knowledge of your entire codebase to give more accurate answers and generate idiomatic code.
```

Author: Do you think it's too much personification to refer to Cody with "who" here? I don't think we've done that before

Reviewer: Yeah, I also had second thoughts about referring to Cody as "who." I think we can use "that" instead.

Author: ```suggestion
Cody is a free, open-source AI coding assistant that writes, understands, and fixes code. Cody is powered by Sourcegraph's <a href="https://docs.sourcegraph.com/cody/explanations/code_graph_context">code graph</a> and uses knowledge of your entire codebase to give more accurate answers and generate idiomatic code.
```

## Main features

Cody's main features include:

<!-- NOTE: These should stay roughly in sync with client/cody/README.md, although these need to be not specific to VS Code. -->

|     Feature     |                                                                                         Description                                                                                         |
| --------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | [Autocomplete](./../capabilities.md#autocomplete)    | Cody makes context-based code autocompletions. Cody can autocomplete single lines, or whole functions, in any programming language, configuration file, or documentation.

Reviewer: ```suggestion
| [Autocomplete](./../capabilities.md#autocomplete)    | Cody makes context-based code autocompletions. Cody can autocomplete single lines or whole functions in any programming language, configuration file, or documentation.
```

> | [Chat](./../capabilities.md#chat)    | Ask Cody questions in the chat view, or inline with code, and it will use Sourcegraph’s code graph to answer using knowledge of your codebase. |

Reviewer: ```suggestion
| [Chat](./../capabilities.md#chat)    | Ask Cody questions in the chat view or inline with code, and it will use Sourcegraph’s code graph to answer using knowledge of your codebase. |
```
